File: src/NubipApp.Models/Lesson.cs
namespace NubipApp.Models
{
    public class Lesson
    {
        public int LessonId { get; set; }
        public int CurriculumId { get; set; }
        public int GroupId { get; set; }
        public int TeacherId { get; set; }
        public int AuditoriumId { get; set; }
        public int TimeId { get; set; }
        public string Day { get; set; }
        public int Week { get; set; }
        public int SubGroup { get; set; }

        public Curriculum Curriculum { get; set; }
        public Group Group { get; set; }
        public Teacher Teacher { get; set; }
        public Auditorium Auditorium { get; set; }
        public Time Time { get; set; }
    }
}

File: src/models/lesson.ts
import mongoose from "mongoose";

let Schema = mongoose.Schema;

let LessonSchema = new Schema({
	lessonId: Number,
	curriculumId: [{
		type: Schema.Types.ObjectId, ref: 'Curriculum'
	}],
	groupId: [{
		type: Schema.Types.ObjectId, ref: 'Group'
	}],
	teacherId: [{
		type: Schema.Types.ObjectId, ref: 'Teacher'
	}],
	auditoriumId: [{
		type: Schema.Types.ObjectId, ref: 'Auditorium'
	}],
	timeId: [{
		type: Schema.Types.ObjectId, ref: 'Time'
	}],
	day: String,
	week: Number,
	subGroup: Number,
});

let Lesson = mongoose.model('Lesson', LessonSchema);

export default Lesson;